pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -597,6 +608,233 @@ def test_optimize(self):
         # check that sync was optimized
         self.assertTrue(optimized)
 
+    def test_sync_advisory_new_version(self):
+        """Sync a repository and re-sync with newer version of Advisory.
+
+        Test if advisory with same ID and pkglist, but newer version is updated.
+
+        `Pulp #4142 <https://pulp.plan.io/issues/4142>`_
+
+        1. Sync rpm-unsigned repository
+        2. Re-sync rpm-advisory-updateversion
+        3. Check if the newer version advisory was synced
+        """
+        body = gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+
+        # sync
+        repo, remote = self.do_test(remote=remote)
+
+        body = gen_rpm_remote(RPM_ADVISORY_UPDATED_VERSION_URL)
+        remote = self.remote_api.create(body)
+
+        # re-sync
+        repo, remote = self.do_test(repo, remote)
+
+        # add resources to clean up
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # check if newer version advisory was added and older removed
+        added_advisories = get_added_content(repo.to_dict())[PULP_TYPE_ADVISORY]
+        added_advisory = [
+            advisory['version']
+            for advisory in added_advisories
+            if advisory['id'] == RPM_ADVISORY_TEST_ID
+        ]
+        removed_advisories = get_removed_content(repo.to_dict())[PULP_TYPE_ADVISORY]
+        removed_advisory = [
+            advisory['version']
+            for advisory in removed_advisories
+            if advisory['id'] == RPM_ADVISORY_TEST_ID
+        ]
+        self.assertGreater(int(added_advisory[0]), int(removed_advisory[0]))
+
+    def test_sync_merge_advisories(self):
+        """Sync two advisories with same ID, version and different pkglist.
+
+        Test if two advisories are merged.
+        """
+        body = gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL)
+        remote = self.remote_api.create(body)
+
+        # sync
+        repo, remote = self.do_test(remote=remote)
+
+        body = gen_rpm_remote(RPM_ADVISORY_DIFFERENT_PKGLIST_URL)
+        remote = self.remote_api.create(body)
+
+        # re-sync
+        repo, remote = self.do_test(repo, remote)
+
+        # add resources to clean up
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # check advisories were merged
+        added_advisories = get_added_content(repo.to_dict())[PULP_TYPE_ADVISORY]
+        added_advisory_pkglist = [
+            advisory['pkglist']
+            for advisory in added_advisories
+            if advisory['id'] == RPM_ADVISORY_TEST_ID
+        ][0]
```

Comment:
Sometimes you use index here, sometimes you use it later and not with list comprehension (see other tests you added), please unify.